ert3/storage/_storage.py

```diff
@@ -131,11 +180,87 @@ def _get_data(workspace: Path, experiment_name: str, record_name: str) -> Any:
     )
 
     if response.status_code == 404:
-        raise KeyError(f"No {record_name} data for experiment: {experiment_name}")
+        raise ert3.exceptions.ElementMissingError(
+            f"No {record_name} data for experiment: {experiment_name}"
+        )
+
+    if response.status_code != 200:
+        raise ert3.exceptions.StorageError(response.text)
 
     return response.content
 
 
+def _response2record(
+    response_content: bytes, record_type: ert3.data.RecordType, realization_id: int
+) -> ert3.data.Record:
+    dataframe = pd.read_csv(
+        io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+    )
+
+    raw_index = tuple(dataframe.columns)
+    if record_type == ert3.data.RecordType.LIST_FLOAT:
+        array_data = tuple(
+            float(dataframe.loc[realization_id][raw_idx]) for raw_idx in raw_index
+        )
+        return ert3.data.Record(data=array_data)
+    elif record_type == ert3.data.RecordType.MAPPING_INT_FLOAT:
+        int_index = tuple(int(e) for e in dataframe.columns)
+        idata = {
+            idx: float(dataframe.loc[realization_id][raw_idx])
+            for raw_idx, idx in zip(raw_index, int_index)
+        }
+        return ert3.data.Record(data=idata)
+    elif record_type == ert3.data.RecordType.MAPPING_STR_FLOAT:
+        str_index = tuple(str(e) for e in dataframe.columns)
+        sdata = {
+            idx: float(dataframe.loc[realization_id][raw_idx])
+            for raw_idx, idx in zip(raw_index, str_index)
+        }
+        return ert3.data.Record(data=sdata)
+    else:
+        raise ValueError(
+            f"Unexpected record type when loading numerical record: {record_type}"
+        )
+
+
+def _get_numerical_data(
+    workspace: Path, experiment_name: str, record_name: str
+) -> ert3.data.EnsembleRecord:
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise ert3.exceptions.NonExistantExperiment(
+            f"Cannot get {record_name} data, no experiment named: {experiment_name}"
+        )
+
+    ensemble_id = experiment["ensembles"][0]  # currently just one ens per exp
+
+    metadata = json.loads(
+        _get_file_data(workspace, experiment_name, record_name + _METADATA)
+    )
+    record_type = ert3.data.RecordType(metadata["record_type"])
+
+    records = []
+    for real_id in range(metadata["ensemble_size"]):
+        response = requests.get(
```

Comment:
Not sure if this helps but we have a specific endpoint for getting all realization data for a response:
`@router.get("/ensembles/{ensemble_id}/responses/{response_name}/data")`

Reply:
Unfortunately this does not apply in this scenario :)